src/django_components/extension.py

```diff
@@ -154,43 +168,159 @@ class OnSlotRenderedContext(NamedTuple):
 
 
-class BaseExtensionClass:
-    """Base class for all extension classes."""
+class ExtensionComponentConfig:
+    """
+    `ExtensionComponentConfig` is the base class for all extension component configs.
+
+    Extensions can define nested classes on the component class,
+    such as [`Component.View`](../api#django_components.Component.View) or
+    [`Component.Cache`](../api#django_components.Component.Cache):
+
+    ```py
+    class MyComp(Component):
+        class View:
+            def get(self, request):
+                ...
+
+        class Cache:
+            ttl = 60
+    ```
+
+    This allows users to configure extension behavior per component.
+
+    Behind the scenes, the nested classes that users define on their components
+    are merged with the extension's "base" class.
+
+    So the example above is the same as:
 
+    ```py
+    class MyComp(Component):
+        class View(ViewExtension.ComponentConfig):
+            def get(self, request):
+                ...
+
+        class Cache(CacheExtension.ComponentConfig):
+            ttl = 60
+    ```
+
+    Where both `ViewExtension.ComponentConfig` and `CacheExtension.ComponentConfig` are
+    subclasses of `ExtensionComponentConfig`.
+    """
+
+    component_cls: Type["Component"]
+    """The [`Component`](../api#django_components.Component) class that this extension is defined on."""
+
+    # TODO_v1 - Remove, superseded by `component_cls`
     component_class: Type["Component"]
-    """The Component class that this extension is defined on."""
+    """The [`Component`](../api#django_components.Component) class that this extension is defined on."""
+
+    component: "Component"
+    """
+    When a [`Component`](../api#django_components.Component) is instantiated,
+    also the nested extension classes (such as `Component.View`) are instantiated,
+    receiving the component instance as an argument.
+
+    This attribute holds the owner [`Component`](../api#django_components.Component) instance
+    that this extension is defined on.
+    """
 
     def __init__(self, component: "Component") -> None:
         self.component = component
 
 
+# TODO_v1 - Delete
+BaseExtensionClass = ExtensionComponentConfig
+"""
+Deprecated. Will be removed in v1.0. Use
+[`ComponentConfig`](../api#django_components.ExtensionComponentConfig) instead.
+"""
+
+
+# TODO_V1 - Delete, meta class was needed only for backwards support for ExtensionClass.
+class ExtensionMeta(type):
+    def __new__(mcs, name: Any, bases: Tuple, attrs: Dict) -> Any:
+        # Rename `ExtensionClass` to `ComponentConfig`
+        if "ExtensionClass" in attrs:
+            attrs["ComponentConfig"] = attrs.pop("ExtensionClass")
+
+        return super().__new__(mcs, name, bases, attrs)
+
+
 # NOTE: This class is used for generating documentation for the extension hooks API.
 #       To be recognized, all hooks must start with `on_` prefix.
-class ComponentExtension:
+class ComponentExtension(metaclass=ExtensionMeta):
```

Comment:
I updated the name of the class inside the Extension that defines how the component-level classes should behave.  Using `ExtensionClass` was confusing. So now instead it's called `ComponentConfig`.

I added a metaclass here to keep things backwards compatible till v1.
The metaclass checks if `ExtensionClass` was defined, and renames it to `ComponentConfig`.